### PR TITLE
Allow nodes to be filtered by status

### LINKED ISSUE #345: Allow nodes to be filtered by status
Nodes should be filterable by their status to make searching easier. A filter selection has to be put next to the search field in the search bar.

### PR DESCRIPTION
**Pull request by julianhaumer**

### CODE REVIEW

File: src/app/editor/components/node-status-filter-selector/node-status-filter-selector.scss [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    &.archived {
+        color: $mesh-node-status-archived;
+        border-color: rgba( $mesh-node-status-archived, 0.4 );
+
+        .status--unpublished {
+            display: block;
+        }
+    }
+
+    & > * + * + * {
```
Comment (julianhaumer) [inline src/app/editor/components/node-status-filter-selector/node-status-filter-selector.scss]: check this selector again :D

File: src/app/editor/components/node-status-filter-selector/node-status-filter-selector.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+                    const selectedStates: DebugElement[] = fixture.debugElement.queryAll(
+                        By.css(`gtx-dropdown-trigger > div.trigger`)
+                    );
+                    expect(selectedStates.length).toEqual(1);
+                }
+            )
+        );
+    });
+
+    describe('Emits a new filter for', () => {
```
Comment (julianhaumer) [inline src/app/editor/components/node-status-filter-selector/node-status-filter-selector.component.spec.ts:118]: test all possible selection (also for tests above)

File: src/app/editor/components/search-bar/search-bar.component.ts [hunk 3]
```diff
@@ -103,11 +110,27 @@ export class SearchBarComponent implements OnInit, OnDestroy {
...
             .map(uuid => this.entities.getTag(uuid))
             .filter(notNullOrUndefined);
+        // filter unknown node statuses and remove duplicates
+        this.searchNodeStatusFilter = [
+            ...new Set((params.get('n') || '').split(',').filter(this.isEMeshNodeStatusString))
+        ];
+
+        // if no filters are set, then it is assumed that no nodes should be filtered (e.g. setting all node statuses as filters)
+        if (this.searchNodeStatusFilter.length === 0) {
+            this.searchNodeStatusFilter = Object.values(EMeshNodeStatusStrings);
```
Comment (julianhaumer) [inline src/app/editor/components/search-bar/search-bar.component.ts]: use nodeStatuses

File: src/app/core/providers/i18n/translations_json/list.translations.json [hunk 4]
```diff
@@ -69,6 +69,13 @@
       "pt": "Editar",
       "hu": "Szerkesztés"
     },
+    "node_status_all": {
+        "en": "All",
+        "de": "Alle",
+        "zh": "All",
```
Comment (julianhaumer) [inline src/app/core/providers/i18n/translations_json/list.translations.json]: add translations for zh, pt, hu

File: src/app/editor/components/container-contents/container-contents.component.ts [hunk 5]
```diff
@@ -81,11 +98,18 @@ export class ContainerContentsComponent implements OnInit, OnDestroy {
...
             )
             .subscribe(({ containerUuid, projectName, language }) => {
+                this.currentLanguage = language;
                 this.listEffects.setActiveContainer(projectName, containerUuid, language);
             });
 
         // get search filter from url parameters
-        const searchParams$ = this.route.queryParamMap.pipe(map(this.extractQueryParams));
+        const searchParams$: Observable<{
+            keyword: string;
```
Comment (julianhaumer) [inline src/app/editor/components/container-contents/container-contents.component.ts]: create an alias

File: src/app/editor/components/container-contents/container-contents.component.ts [hunk 6]
```diff
@@ -161,9 +187,29 @@ export class ContainerContentsComponent implements OnInit, OnDestroy {
             )
         );
 
-        this.searching$ = searchParams$.pipe(map(({ keyword, tags }) => keyword !== '' || tags !== ''));
+        this.searching$ = searchParams$.pipe(
+            tap(({ nodeStatusFilter }) => {
```
Comment (julianhaumer) [inline src/app/editor/components/container-contents/container-contents.component.ts:187]: maybe util 🤷

File: src/app/shared/components/node-status/node-status.component.html [hunk 7]
```diff
@@ -9,31 +9,31 @@
     <span class="text-uppercase">{{ 'common.' + nodeLabel | i18n }}</span>
     <!-- DRAFT -->
     <p [@animNgIfation]
-       *ngIf="(hasHover | async) && !nodeIsPublished && getNodeHasBeenEdited()">
+       *ngIf="(hasHover | async) && (node | nodeStatus: current) === nodeStatusStringsEnum.DRAFT">
```
Comment (julianhaumer) [inline src/app/shared/components/node-status/node-status.component.html:12]: switch case ?

**Comment by apagtxmo** [comment]
Did a Review with @julianhaumer . He will check some optional changes we noted while reviewing. Eveything else looks good and can be merged

**Pull request merged**